✨ feat(calendar): add basic calendar structure and styling

de-vasta opened the pull request:
- create calendar grid with days
- implement basic styling using scss variables and functions
- add hover effect for day elements
- define flexible calendar starting day and month length

de-vasta committed:
✨ feat(calendar): add basic calendar structure and styling
- create calendar grid with days
- implement basic styling using scss variables and functions
- add hover effect for day elements
- define flexible calendar starting day and month length

mateacademy-ai-mentor commented on src/styles/calendar.scss:
```diff
@@ -0,0 +1,55 @@
+    background-color: $day-bg-color;
+    font-size: $day-font-size;
+
+    transition: transform $day-transition-time;
```
The requirement is that all changes on hover should be animated. Currently, the transition only applies to the `transform` property. The `background-color` change should also be included in the transition.

mateacademy-ai-mentor commented on src/styles/calendar.scss:
```diff
@@ -0,0 +1,55 @@
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: $day-bg-color-hover;
```
The task requires the hovered cell to have a background color of `#FFBFCB`. The variable `$day-bg-color-hover` is currently set to a different color in `variables.scss`. Please update the variable's value to match the requirement.

mateacademy-ai-mentor commented on src/styles/calendar.scss:
```diff
@@ -0,0 +1,55 @@
+
+  @each $week-day, $week-days-left in $calendar-1st-row {
+    &--start-day-#{$week-day} {
+      & :first-child {
```
This selector targets any descendant that is a first child. According to BEM best practices, it's better to use a direct child selector (`>`) to be more specific and avoid unintended styling on nested elements. This also relates to checklist item #3 regarding BEM 'Typical Mistakes'.

Change `& :first-child` to `& > :first-child`.

mateacademy-ai-mentor commented on src/styles/calendar.scss:
```diff
@@ -0,0 +1,55 @@
+    }
+  }
+
+  @for $i from 1 through ($days + 1) {
```
This loop runs from 1 up to `($days + 1)`, which evaluates to 32 since `$days` is 31. However, there are only 31 day elements in the HTML. The loop should run up to `$days` to avoid creating a rule for a non-existent 32nd element.

mateacademy-ai-mentor commented on src/styles/main.css:
```diff
@@ -0,0 +1,18 @@
+
+body {
+  margin: 0;
+}
```
The `body` styles are incomplete. To center the calendar on the page as required, you need to add flexbox properties here, such as `display: flex`, `justify-content: center`, `align-items: center`, and `height: 100vh`.

mateacademy-ai-mentor commented on src/styles/main.css:
```diff
@@ -0,0 +1,18 @@
+
+.month {
+  display: flex;
+}
```
The `.month` class is not used in your `index.html` file. This style rule appears to be unused and should be removed to keep the code clean.

mateacademy-ai-mentor commented on src/styles/main.css:
```diff
@@ -0,0 +1,18 @@
+.calendar__day {
+  width: 100px;
+  height: 100px;
+  color: #eee;
```
The color for the day numbers is set to `#eee`. Since the background color of the days is also `#eee` (from `calendar.scss`), this makes the numbers invisible. Please ensure the numbers have a contrasting color so they are visible.

mateacademy-ai-mentor commented on src/styles/variables.scss:
```diff
@@ -0,0 +1,21 @@
+$font-family: Arial;
+$days: 31;
+$day-bg-color: #eee;
+$day-bg-color-hover: #f1e5e7;
```
The required background color for a hovered cell is pink (`#FFBFCB`). The current value does not match the requirement.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great job on the overall structure and use of SCSS features. However, a few changes are needed to meet all the requirements. The hover effect is incomplete: the background color should be pink (`#FFBFCB`) as specified in `src/styles/variables.scss`, and the `transition` in `src/styles/calendar.scss` must also apply to the `background-color` change. Additionally, the `@for` loop for day numbers iterates one time too many, and the selector for the first day (`& :first-child`) should be more specific (`& > :first-child`) to align with BEM best practices. Finally, please ensure your compiled CSS correctly centers the calendar and that the day numbers are visible against their background. After fixing these points, your solution will be perfect!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
💄 style(calendar): improve calendar styling and remove unused css

- remove generated css and map files
- add hover effect to calendar days
- adjust calendar day calculation

de-vasta committed:
💄 style(calendar): improve calendar styling and remove unused css

- remove generated css and map files
- add hover effect to calendar days
- adjust calendar day calculation

mateacademy-ai-mentor commented on src/styles/calendar.scss:
```diff
@@ -0,0 +1,58 @@
+.calendar {
+  margin: 0 auto;
```
This `margin` property is redundant for horizontal centering. The parent `body` element is already a flex container with `justify-content: center`, which takes care of centering the calendar. You can safely remove this line.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this task! I am approving your solution as you've successfully met all the requirements. The calendar is perfectly centered, the SCSS is well-structured, and all modifiers work as expected. You've done a great job addressing all the previous feedback.

As a minor suggestion for future improvement, the `margin` property in `src/styles/calendar.scss` is redundant for horizontal centering since the parent `body` already handles it with flexbox. Removing it would make the code even cleaner, but this is not a blocker. Overall, this is a fantastic submission!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

de-vasta committed:
Merge branch 'calendar' of https://github.com/de-vasta/layout_calendar into calendar
